src/Nethermind/Nethermind.BlockValidation/Handlers/ValidateBuilderSubmissionHandler.cs

```diff
@@ -0,0 +1,331 @@
+
+        return true;
+    }
+
+    private BlockProcessor CreateBlockProcessor(IWorldState stateProvider)
+    {
+        return new BlockProcessor(
+            _txProcessingEnv.SpecProvider,
+            _blockValidator,
+            new Consensus.Rewards.RewardCalculator(_txProcessingEnv.SpecProvider),
+            new BlockProcessor.BlockValidationTransactionsExecutor(_txProcessingEnv.TransactionProcessor, stateProvider),
+            stateProvider,
+            _receiptStorage,
+            new BlockhashStore(_txProcessingEnv.SpecProvider, stateProvider),
+            _txProcessingEnv.LogManager,
+            new WithdrawalProcessor(stateProvider, _txProcessingEnv.LogManager!),
+            new ReceiptsRootCalculator()
+        );
+    }
+
+    private static void FinalizeStateAndBlock(IWorldState stateProvider, Block processedBlock, IReleaseSpec currentSpec, Block currentBlock, IBlockTree blockTree)
+    {
+        stateProvider.StateRoot = processedBlock.StateRoot!;
+        stateProvider.Commit(currentSpec);
+        stateProvider.CommitTree(currentBlock.Number);
+        blockTree.SuggestBlock(processedBlock, BlockTreeSuggestOptions.ForceSetAsMain);
+        blockTree.UpdateHeadBlock(processedBlock.Hash!);
+    }
+
+
+    private bool ValidatePayload(Block block, Address feeRecipient, UInt256 expectedProfit, long registerGasLimit, bool useBalanceDiffProfit, bool excludeWithdrawals, out string? error)
+    {
+        if (!HeaderValidator.ValidateHash(block.Header))
+        {
+            error = $"Invalid block header hash {block.Header.Hash}";
+            return false;
+        }
+
+        if (!_blockTree.IsBetterThanHead(block.Header))
+        {
+            error = $"Block {block.Header.Hash} is not better than head";
+            return false;
+        }
+
+        BlockHeader? parentHeader = _blockTree.FindHeader(block.ParentHash!, BlockTreeLookupOptions.DoNotCreateLevelIfMissing);
+
+        if (parentHeader is null)
+        {
+            error = $"Parent header {block.ParentHash} not found";
+            return false;
+        }
+
+        long calculatedGasLimit = _gasLimitCalculator.GetGasLimit(parentHeader, registerGasLimit);
+
+        if (calculatedGasLimit != block.Header.GasLimit)
+        {
+            error = $"Gas limit mismatch. Expected {calculatedGasLimit} but got {block.Header.GasLimit}";
+            return false;
+        }
+
+        IReadOnlyTxProcessingScope processingScope = _txProcessingEnv.Build(parentHeader.StateRoot!);
+        IWorldState currentState = processingScope.WorldState;
+
+        UInt256 feeRecipientBalanceBefore = currentState.GetBalance(feeRecipient);
+
+        BlockProcessor blockProcessor = CreateBlockProcessor(currentState);
+
+        List<Block> suggestedBlocks = [block];
+        BlockReceiptsTracer blockReceiptsTracer = new();
+
+        ProcessingOptions processingOptions = new ProcessingOptions();
```

Comment:
This is not correct. We want: `ReadOnlyChain` and maybe `ForceProcessing`, `IgnoreParentNotOnMainChain`, `DoNotVerifyNonce` (if all that would be all under `Trace`)